libbeat/cmd/instance/beat_integration_test.go

```diff
@@ -0,0 +1,130 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package instance_test
+
+import (
+	"encoding/json"
+	"flag"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/cmd/instance"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/mock"
+)
+
+type mockbeat struct {
+	done     chan struct{}
+	initDone chan struct{}
+}
+
+func (mb mockbeat) Run(b *beat.Beat) error {
+	client, err := b.Publisher.Connect()
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(1 * time.Second)
+	go func() {
+		// unblocks mb.waitUntilRunning
+		close(mb.initDone)
+		for {
+			select {
+			case <-ticker.C:
+				client.Publish(beat.Event{
+					Timestamp: time.Now(),
+					Fields: common.MapStr{
+						"type":    "mock",
+						"message": "Mockbeat is alive!",
+					},
+				})
+			case <-mb.done:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+
+	<-mb.done
+	return nil
+}
+
+func (mb mockbeat) waitUntilRunning() {
+	<-mb.initDone
+}
+
+func (mb mockbeat) Stop() {
+	close(mb.done)
+}
+
+func TestMonitoringNameFromConfig(t *testing.T) {
+	mockBeat := mockbeat{
+		done:     make(chan struct{}),
+		initDone: make(chan struct{}),
+	}
+	wg := sync.WaitGroup{}
```

Comment:
Nit: we usually write `var wg sync.WaitGroup` in this codebase.

Reply:
Done on 7ccb3aba12